Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM rust:1.69-bullseye AS chef
+RUN rustup update stable --no-self-update \
+    && rustup default stable \
+    && rustup target add wasm32-wasi \
+    && rustup target add wasm32-unknown-unknown
+RUN mkdir -p /usr/local/protobuf \
+    && wget https://github.com/protocolbuffers/protobuf/releases/download/v22.3/protoc-22.3-linux-x86_64.zip \
+    && unzip protoc-22.3-linux-x86_64.zip -d /usr/local/protobuf \
+    && rm protoc-22.3-linux-x86_64.zip
+ENV PATH="${PATH}:/usr/local/protobuf/bin"
+RUN cargo install cargo-chef --locked
+
+FROM chef AS planner
+WORKDIR /usr/src/bytecodealliance/registry
+COPY . .
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder
+WORKDIR /usr/src/bytecodealliance/registry
+COPY --from=planner /usr/src/bytecodealliance/registry/recipe.json ./
+RUN cargo chef cook --release --workspace --recipe-path recipe.json
+COPY . .
+RUN cargo build --release --workspace
+
+FROM gcr.io/distroless/cc AS warg-server
```

Comment:
For debugging purposes it can be quite helpful to have a shell and debug tools available on server containers. Is there another way to do that with distroless images?

Reply:
Yes, I'd suggest we could add an `ARG` to do a [debug variant](https://github.com/GoogleContainerTools/distroless#debug-images) which is in turn used in the `docker-compose`, but otherwise the `Dockerfile` would build a release version.

Reply:
Probably a separate target would be better, that we can also do something like make it remotely debuggable, e.g., debug build and include `gdbserver` in the container and use it for the entrypoint? Follow up PR I'd say...

Reply:
I created a `warg-server-debug` Dockerfile target for convenience if a developer wants have more tools on the vm than just `bash` and other Linux basics.